### PR TITLE
Polish and improve documentation

### PR DESCRIPTION
**Pull request by mssun**
## Description
  - Polish and improve documentation
  - Add document for built-in functions

### CODE REVIEW

File: third_party/README.md [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
-from external sources such as [crates.io](https://crates.io).
+# Third-Party Dependency Vendoring
 
-## To Add A New Vendored Dependency
+For Teaclave, since all dependencies are part of trusted computing base (TCB),
+in order to ease auditing, ensure product stability, as well as reduce the
+possibility of the [supply chain attack](https://en.wikipedia.org/wiki/Supply_chain_attack),
+we vendored all TEE dependencies here. During the build process, both the
+untrusted (i.e., the app part) and trusted components (i.e., the enclave part)
+will only consumes packages from this designated repository and will not
```
Comment (m4sterchain) [inline third_party/README.md:12]: will only comsumes -> only comsume

File: third_party/README.md [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
-In order to ease auditing, ensure product stability, as well as reduce the
-possibility of the [supply chain
-attack](https://en.wikipedia.org/wiki/Supply_chain_attack), we vendored all TEE
-dependencies here.  During the build process, the trusted components will only
-consumes packages from this designated repository and will not download any code
-from external sources such as [crates.io](https://crates.io).
+# Third-Party Dependency Vendoring
 
-## To Add A New Vendored Dependency
+For Teaclave, since all dependencies are part of trusted computing base (TCB),
```
Comment (m4sterchain) [inline third_party/README.md:7]: all SGX/Enclave dependencies

File: third_party/README.md [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-[crates-sgx/README.txt](https://github.com/mesalock-linux/crates-sgx/blob/master/README.txt)).
-2. Run ``cargo build`` and ensure that it passes.
-3. Run ``cargo vendor`` and update the config file (e.g. crates-sgx/config).
-   You may also utilize
-[crates-sgx/Makefile](https://github.com/mesalock-linux/crates-sgx/blob/master/Makefile)
-for automation.
-4. ``git add/commit`` the changes of
-   Cargo.toml/Cargo.lock/config/README.txt/vendor and submit a pull request.
+Basically, we have these submodules:
+  - `crates-io`: Rust crates vendored from crates.io.
```
Comment (m4sterchain) [inline third_party/README.md:17]: for non-SGX components.

File: function/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+---
+
+# Built-in Functions
+
+Since Teaclave is a FaaS-like platform, users can define and register customized
+functions (e.g., written in Python). To make data computation more easier and
+faster (in native speed), the platform also provide some commonly used functions
+written in Rust. We call them built-in functions. These functions can be
+selectively compiled in the *built-in executor* with a "builtin" prefix in the
+function names.
```
Comment (m4sterchain) [inline function/README.md:12]: Users can selectively invoke these build-in functions by their names.

**Comment by mssun** [comment]
Fixed.

**Pull request merged**